webapp/src/components/AssessmentDetailPage.tsx

```diff
@@ -0,0 +1,329 @@
+    submissionIdNumber < 1
+  ) {
+    return (
+      <Alert severity="error">
+        <AlertTitle>Sorry!</AlertTitle>
+        The assessment ID and submission ID must be valid.
+      </Alert>
+    );
+  }
+
+  if (
+    !assessment ||
+    (assessment && typeof assessment.curriculum_assessment === 'undefined') ||
+    (assessment &&
+      typeof assessment.curriculum_assessment.questions === 'undefined') ||
+    (Array.isArray(assessment.curriculum_assessment.questions) &&
+      assessment.curriculum_assessment.questions.length === 0) ||
+    typeof assessment.submission.responses === 'undefined' ||
+    (Array.isArray(assessment.submission.responses) &&
+      assessment.submission.responses.length === 0) ||
+    !endTime
+  ) {
+    return (
+      <Alert severity="error">
+        <AlertTitle>Sorry!</AlertTitle>
+        <p>There was a problem loading this assessment or submission.</p>
+      </Alert>
+    );
+  }
+
+  return (
+    <Container>
+      <Grid container spacing={2}>
+        <Grid item xs={12} md={6}>
+          <h1>{assessment.curriculum_assessment.title}</h1>
+        </Grid>
+
+        <Grid item xs={12} md={12}>
+          <Grid container spacing={2}>
+            <Grid item xs={12} md={3}>
+              <AssessmentMetadataBar
+                assessment={assessment}
+                secondsRemaining={secondsRemaining}
+                endTime={endTime}
+                submissionDisabled={submissionDisabled}
+              />
+            </Grid>
+
+            <Grid
+              id="assessment_display"
+              item
+              xs={12}
+              md={7.5}
+              sx={{
+                height: '75vh',
+                overflowX: 'auto',
+                overflowY: 'scroll',
+                backgroundColor: '#fafafa',
+                boxShadow: '0 0 0.25em rgba(0,0,0,0.35)',
+              }}
+            >
+              <AssessmentDisplay
+                assessment={assessment}
+                handleUpdatedResponse={handleUpdatedResponse}
+                questionsDisabled={submissionDisabled}
+              />
+            </Grid>
+
+            <Grid item xs={12} md={1.5}>
+              <AssessmentSubmitBar
+                assessment={assessment}
+                numOfAnsweredQuestions={numOfAnsweredQuestions}
+                setShowSubmitDialog={setShowSubmitDialog}
+                submitButtonDisabled={submissionDisabled}
+              />
+            </Grid>
+          </Grid>
+        </Grid>
+      </Grid>
+      <Dialog
+        open={showSubmitDialog}
+        onClose={() => setShowSubmitDialog(false)}
+      >
+        <DialogTitle>
+          Are you sure you would like to submit this assessment?
+        </DialogTitle>
+        {numOfAnsweredQuestions <
+          assessment.curriculum_assessment.questions!.length && (
+          <DialogContent>
+            <DialogContentText>
+              {`You have only responded to ${numOfAnsweredQuestions} out of ` +
```

Comment:
I like that it changes based on if the user has answered all the questions or not! I was about to give some feedback that it should say "question" at the end of the message, but it looks like that was already addressed. Really well done!!!